### ISSUE TITLE
SPV messages while running a Neutrino node

### ISSUE THREAD

**Issue by nkohen** [bitcoin-s/bitcoin-s#1093]
I have a node running with this in the `bitcoin-s.conf`:
```
    node {
        mode = neutrino # neutrino, spv

        peers = ["localhost:18444"]
    }

```
and I ran `getnewaddress` and then ran `bitcoin-cli generateToAddress 10 <the address>` and got the following logs:
```
2020-02-03T19:48:59 DEBUG [P2P] Received message=inv from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:48:59 INFO [P2P] Received inv=InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(9abd99a32b9cebd52ca78f4c45bc780fceb95b0df947e2cbb9ea6c8b5686ea13)))
2020-02-03T19:48:59 DEBUG [P2P] Sending msg=getdata to peer=localhost/127.0.0.1:18444
2020-02-03T19:48:59 DEBUG [P2P] Received 362 TCP bytes
2020-02-03T19:48:59 DEBUG [P2P] Parsed 3 message(s) from bytes: NetworkMessage(NetworkHeader(RegTest,merkleblock,UInt32Impl(119),ByteVector(4 bytes, 0xd6d79fa2)), MerkleBlockMessage(MerkleBlockImpl(BlockHeaderImpl(Int32Impl(536870912),DoubleSha256Digest(f97a5d08efac380865ad26a340b579118b90b9ac68f065c20d6caca4c0a5ad72),DoubleSha256Digest(25e8018d9895f6784431f8b1236ad21cde7f7dc9f30f1032233da01998522a5a),UInt32Impl(1580759339),UInt32Impl(545259519),UInt32Impl(3)),UInt32Impl(1),PartialMerkleTreeImpl(Leaf(DoubleSha256Digest(25e8018d9895f6784431f8b1236ad21cde7f7dc9f30f1032233da01998522a5a)),UInt32Impl(1),BitVector(8 bits, 0x80),List(DoubleSha256Digest(25e8018d9895f6784431f8b1236ad21cde7f7dc9f30f1032233da01998522a5a)))))), NetworkMessage(NetworkHeader(RegTest,tx,UInt32Impl(134),ByteVector(4 bytes, 0x25e8018d)), TransactionMessage(DoubleSha256BDigestBE(5a2a529819a03d2332100ff3c97d7fde1cd26a23b1f8314478f695988d01e825))), NetworkMessage(NetworkHeader(RegTest,inv,UInt32Impl(37),ByteVector(4 bytes, 0x1f948d67)), InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(500f178982b4a5dbced8b4f11dff2de517934c8ca18d4cd1a67d67b02d715a0f))))
2020-02-03T19:48:59 DEBUG [P2P] Unaligned bytes after this: 0
2020-02-03T19:48:59 DEBUG [P2P] Received message=merkleblock from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:48:59 DEBUG [P2P] Received merkle block, expecting 1 TX(s)
2020-02-03T19:48:59 DEBUG [P2P] Received message=tx from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:48:59 DEBUG [P2P] Transaction=DoubleSha256BDigestBE(5a2a529819a03d2332100ff3c97d7fde1cd26a23b1f8314478f695988d01e825) matched merkleBlock=DoubleSha256BDigestBE(13ea86568b6ceab9cbe247f90d5bb9ce0f78bc454c8fa72cd5eb9c2ba399bd9a)
2020-02-03T19:48:59 DEBUG [P2P] We've received all transactions (1) for merkleBlock=DoubleSha256BDigestBE(13ea86568b6ceab9cbe247f90d5bb9ce0f78bc454c8fa72cd5eb9c2ba399bd9a)
2020-02-03T19:48:59 DEBUG [P2P] Received message=inv from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:48:59 INFO [P2P] Received inv=InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(500f178982b4a5dbced8b4f11dff2de517934c8ca18d4cd1a67d67b02d715a0f)))
2020-02-03T19:48:59 DEBUG [P2P] Sending msg=getdata to peer=localhost/127.0.0.1:18444
2020-02-03T19:48:59 DEBUG [P2P] Received 362 TCP bytes
2020-02-03T19:48:59 DEBUG [P2P] Parsed 3 message(s) from bytes: NetworkMessage(NetworkHeader(RegTest,merkleblock,UInt32Impl(119),ByteVector(4 bytes, 0x258447db)), MerkleBlockMessage(MerkleBlockImpl(BlockHeaderImpl(Int32Impl(536870912),DoubleSha256Digest(9fe219ac2be32e032ce0c3f339751566ac99f151c3913208a142e046c9b82b4f),DoubleSha256Digest(45b4e051c62b2ce07bbeba58d16c8da5d38c1e38baebea3ba22df27ed680ae80),UInt32Impl(1580759339),UInt32Impl(545259519),UInt32Impl(1)),UInt32Impl(1),PartialMerkleTreeImpl(Leaf(DoubleSha256Digest(45b4e051c62b2ce07bbeba58d16c8da5d38c1e38baebea3ba22df27ed680ae80)),UInt32Impl(1),BitVector(8 bits, 0x80),List(DoubleSha256Digest(45b4e051c62b2ce07bbeba58d16c8da5d38c1e38baebea3ba22df27ed680ae80)))))), NetworkMessage(NetworkHeader(RegTest,tx,UInt32Impl(134),ByteVector(4 bytes, 0x45b4e051)), TransactionMessage(DoubleSha256BDigestBE(80ae80d67ef22da23beaebba381e8cd3a58d6cd158babe7be02c2bc651e0b445))), NetworkMessage(NetworkHeader(RegTest,inv,UInt32Impl(37),ByteVector(4 bytes, 0xb6826290)), InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(54b9e35ef6aa00a5a2efe1edf39adba5eeba30c64a0404cff0566debac6ff34f))))
2020-02-03T19:48:59 DEBUG [P2P] Unaligned bytes after this: 0
2020-02-03T19:48:59 DEBUG [P2P] Received message=merkleblock from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:48:59 DEBUG [P2P] Received merkle block, expecting 1 TX(s)
2020-02-03T19:48:59 DEBUG [P2P] Received message=tx from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:48:59 DEBUG [P2P] Transaction=DoubleSha256BDigestBE(80ae80d67ef22da23beaebba381e8cd3a58d6cd158babe7be02c2bc651e0b445) matched merkleBlock=DoubleSha256BDigestBE(0f5a712db0677da6d14c8da18c4c9317e52dff1df1b4d8cedba5b48289170f50)
2020-02-03T19:48:59 DEBUG [P2P] We've received all transactions (1) for merkleBlock=DoubleSha256BDigestBE(0f5a712db0677da6d14c8da18c4c9317e52dff1df1b4d8cedba5b48289170f50)
2020-02-03T19:48:59 DEBUG [P2P] Received message=inv from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:48:59 INFO [P2P] Received inv=InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(54b9e35ef6aa00a5a2efe1edf39adba5eeba30c64a0404cff0566debac6ff34f)))
2020-02-03T19:48:59 DEBUG [P2P] Sending msg=getdata to peer=localhost/127.0.0.1:18444
2020-02-03T19:48:59 DEBUG [P2P] Received 301 TCP bytes
2020-02-03T19:48:59 DEBUG [P2P] Parsed 2 message(s) from bytes: NetworkMessage(NetworkHeader(RegTest,merkleblock,UInt32Impl(119),ByteVector(4 bytes, 0x19dbaf0b)), MerkleBlockMessage(MerkleBlockImpl(BlockHeaderImpl(Int32Impl(536870912),DoubleSha256Digest(8e74563439558706bcec589f97e1406934eafef0ce6c57b861ffb36915b66653),DoubleSha256Digest(61c65b27d5260745c23e437178ae5448f1c6fcb2aed588b7a5edc1e644c0d12f),UInt32Impl(1580759339),UInt32Impl(545259519),UInt32Impl(1)),UInt32Impl(1),PartialMerkleTreeImpl(Leaf(DoubleSha256Digest(61c65b27d5260745c23e437178ae5448f1c6fcb2aed588b7a5edc1e644c0d12f)),UInt32Impl(1),BitVector(8 bits, 0x80),List(DoubleSha256Digest(61c65b27d5260745c23e437178ae5448f1c6fcb2aed588b7a5edc1e644c0d12f)))))), NetworkMessage(NetworkHeader(RegTest,tx,UInt32Impl(134),ByteVector(4 bytes, 0x61c65b27)), TransactionMessage(DoubleSha256BDigestBE(2fd1c044e6c1eda5b788d5aeb2fcc6f14854ae7871433ec2450726d5275bc661)))
2020-02-03T19:48:59 DEBUG [P2P] Unaligned bytes after this: 0
2020-02-03T19:48:59 DEBUG [P2P] Received message=merkleblock from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:48:59 DEBUG [P2P] Received merkle block, expecting 1 TX(s)
2020-02-03T19:48:59 DEBUG [P2P] Received message=tx from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:48:59 DEBUG [P2P] Transaction=DoubleSha256BDigestBE(2fd1c044e6c1eda5b788d5aeb2fcc6f14854ae7871433ec2450726d5275bc661) matched merkleBlock=DoubleSha256BDigestBE(4ff36faceb6d56f0cf04044ac630baeea5db9af3ede1efa2a500aaf65ee3b954)
2020-02-03T19:48:59 DEBUG [P2P] We've received all transactions (1) for merkleBlock=DoubleSha256BDigestBE(4ff36faceb6d56f0cf04044ac630baeea5db9af3ede1efa2a500aaf65ee3b954)
[INFO] [02/03/2020 13:49:12.019] [bitcoin-s-akka.actor.default-dispatcher-5] [akka.actor.ActorSystemImpl(bitcoin-s)] http-rpc-server: Response for
  Request : HttpRequest(HttpMethod(POST),http://localhost:9999/,List(Timeout-Access: <function1>, Accept-Encoding: gzip, deflate, User-Agent: Java/1.8.0_201, Host: localhost:9999, Accept: text/html, image/gif, image/jpeg, */*;q=0.2, */*;q=0.2, Connection: keep-alive),HttpEntity.Strict(application/json,79 bytes total),HttpProtocol(HTTP/1.1))
  Response: Complete(HttpResponse(200 OK,List(),HttpEntity.Strict(application/json,25 bytes total),HttpProtocol(HTTP/1.1)))
2020-02-03T19:49:19 DEBUG [P2P] Received 122 TCP bytes
2020-02-03T19:49:19 DEBUG [P2P] Parsed 2 message(s) from bytes: NetworkMessage(NetworkHeader(RegTest,inv,UInt32Impl(37),ByteVector(4 bytes, 0x63eef2b2)), InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(f8fb8fe662bacb879de0e3a4d1306080221e0c7ce493cd1075c44d3128541f2b)))), NetworkMessage(NetworkHeader(RegTest,inv,UInt32Impl(37),ByteVector(4 bytes, 0x38b73c5a)), InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(cc2e0d82bc76c8c97882ba583d4d9d5c119d4213e6d4b121c9d556e449533976))))
2020-02-03T19:49:19 DEBUG [P2P] Unaligned bytes after this: 0
2020-02-03T19:49:19 DEBUG [P2P] Received message=inv from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:49:19 INFO [P2P] Received inv=InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(f8fb8fe662bacb879de0e3a4d1306080221e0c7ce493cd1075c44d3128541f2b)))
2020-02-03T19:49:19 DEBUG [P2P] Sending msg=getdata to peer=localhost/127.0.0.1:18444
2020-02-03T19:49:19 DEBUG [P2P] Received message=inv from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:49:19 INFO [P2P] Received inv=InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(cc2e0d82bc76c8c97882ba583d4d9d5c119d4213e6d4b121c9d556e449533976)))
2020-02-03T19:49:19 DEBUG [P2P] Sending msg=getdata to peer=localhost/127.0.0.1:18444
2020-02-03T19:49:19 DEBUG [P2P] Received 663 TCP bytes
2020-02-03T19:49:19 DEBUG [P2P] Parsed 5 message(s) from bytes: NetworkMessage(NetworkHeader(RegTest,merkleblock,UInt32Impl(119),ByteVector(4 bytes, 0x94e86eef)), MerkleBlockMessage(MerkleBlockImpl(BlockHeaderImpl(Int32Impl(536870912),DoubleSha256Digest(3d5cfcee397f954db237a7228fd796f9ec4145a7dcdf81433097aaad9340207b),DoubleSha256Digest(d8f731e1fed6c3aefd79a6256c38fdbddb375beb9663b3765b4b17c3c64b1cb5),UInt32Impl(1580759359),UInt32Impl(545259519),UInt32Impl(0)),UInt32Impl(1),PartialMerkleTreeImpl(Leaf(DoubleSha256Digest(d8f731e1fed6c3aefd79a6256c38fdbddb375beb9663b3765b4b17c3c64b1cb5)),UInt32Impl(1),BitVector(8 bits, 0x80),List(DoubleSha256Digest(d8f731e1fed6c3aefd79a6256c38fdbddb375beb9663b3765b4b17c3c64b1cb5)))))), NetworkMessage(NetworkHeader(RegTest,tx,UInt32Impl(134),ByteVector(4 bytes, 0xd8f731e1)), TransactionMessage(DoubleSha256BDigestBE(b51c4bc6c3174b5b76b36396eb5b37dbbdfd386c25a679fdaec3d6fee131f7d8))), NetworkMessage(NetworkHeader(RegTest,inv,UInt32Impl(37),ByteVector(4 bytes, 0xcf1a7704)), InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(2df25788d1394596ebad2f25305c43dcdb687b063fbdde81c771f119ccf6e262)))), NetworkMessage(NetworkHeader(RegTest,merkleblock,UInt32Impl(119),ByteVector(4 bytes, 0xb2d4b36a)), MerkleBlockMessage(MerkleBlockImpl(BlockHeaderImpl(Int32Impl(536870912),DoubleSha256Digest(f8fb8fe662bacb879de0e3a4d1306080221e0c7ce493cd1075c44d3128541f2b),DoubleSha256Digest(0553799ede95ed5e85a75e30134a0bd02e0478cc0da7649798f78eff558971b3),UInt32Impl(1580759359),UInt32Impl(545259519),UInt32Impl(0)),UInt32Impl(1),PartialMerkleTreeImpl(Leaf(DoubleSha256Digest(0553799ede95ed5e85a75e30134a0bd02e0478cc0da7649798f78eff558971b3)),UInt32Impl(1),BitVector(8 bits, 0x80),List(DoubleSha256Digest(0553799ede95ed5e85a75e30134a0bd02e0478cc0da7649798f78eff558971b3)))))), NetworkMessage(NetworkHeader(RegTest,tx,UInt32Impl(134),ByteVector(4 bytes, 0x0553799e)), TransactionMessage(DoubleSha256BDigestBE(b3718955ff8ef7989764a70dcc78042ed00b4a13305ea7855eed95de9e795305)))
2020-02-03T19:49:19 DEBUG [P2P] Unaligned bytes after this: 0
2020-02-03T19:49:19 DEBUG [P2P] Received message=merkleblock from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:49:19 DEBUG [P2P] Received merkle block, expecting 1 TX(s)
2020-02-03T19:49:19 DEBUG [P2P] Received message=tx from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:49:19 DEBUG [P2P] Transaction=DoubleSha256BDigestBE(b51c4bc6c3174b5b76b36396eb5b37dbbdfd386c25a679fdaec3d6fee131f7d8) matched merkleBlock=DoubleSha256BDigestBE(2b1f5428314dc47510cd93e47c0c1e22806030d1a4e3e09d87cbba62e68ffbf8)
2020-02-03T19:49:19 DEBUG [P2P] We've received all transactions (1) for merkleBlock=DoubleSha256BDigestBE(2b1f5428314dc47510cd93e47c0c1e22806030d1a4e3e09d87cbba62e68ffbf8)
2020-02-03T19:49:19 DEBUG [P2P] Received message=inv from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:49:19 INFO [P2P] Received inv=InventoryMessage(1 inv(s), Inventory(MsgBlock,DoubleSha256Digest(2df25788d1394596ebad2f25305c43dcdb687b063fbdde81c771f119ccf6e262)))
2020-02-03T19:49:19 DEBUG [P2P] Sending msg=getdata to peer=localhost/127.0.0.1:18444
2020-02-03T19:49:19 DEBUG [P2P] Received message=merkleblock from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:49:19 DEBUG [P2P] Received merkle block, expecting 1 TX(s)
2020-02-03T19:49:19 DEBUG [P2P] Received message=tx from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:49:19 DEBUG [P2P] Transaction=DoubleSha256BDigestBE(b3718955ff8ef7989764a70dcc78042ed00b4a13305ea7855eed95de9e795305) matched merkleBlock=DoubleSha256BDigestBE(76395349e456d5c921b1d4e613429d115c9d4d3d58ba8278c9c876bc820d2ecc)
2020-02-03T19:49:19 DEBUG [P2P] We've received all transactions (1) for merkleBlock=DoubleSha256BDigestBE(76395349e456d5c921b1d4e613429d115c9d4d3d58ba8278c9c876bc820d2ecc)
2020-02-03T19:49:19 DEBUG [P2P] Received 301 TCP bytes
2020-02-03T19:49:19 DEBUG [P2P] Parsed 2 message(s) from bytes: NetworkMessage(NetworkHeader(RegTest,merkleblock,UInt32Impl(119),ByteVector(4 bytes, 0x0e2b4883)), MerkleBlockMessage(MerkleBlockImpl(BlockHeaderImpl(Int32Impl(536870912),DoubleSha256Digest(0a640c484eb5d4bb48fafc1e3ef8366a26c7f93fd41f689a24077c0083b78c00),DoubleSha256Digest(c6a6c9c8668713fe66a76a841c816d07ce837538c921daf5f53ebbf4e235c650),UInt32Impl(1580759359),UInt32Impl(545259519),UInt32Impl(0)),UInt32Impl(1),PartialMerkleTreeImpl(Leaf(DoubleSha256Digest(c6a6c9c8668713fe66a76a841c816d07ce837538c921daf5f53ebbf4e235c650)),UInt32Impl(1),BitVector(8 bits, 0x80),List(DoubleSha256Digest(c6a6c9c8668713fe66a76a841c816d07ce837538c921daf5f53ebbf4e235c650)))))), NetworkMessage(NetworkHeader(RegTest,tx,UInt32Impl(134),ByteVector(4 bytes, 0xc6a6c9c8)), TransactionMessage(DoubleSha256BDigestBE(50c635e2f4bb3ef5f5da21c9387583ce076d811c846aa766fe138766c8c9a6c6)))
2020-02-03T19:49:19 DEBUG [P2P] Unaligned bytes after this: 0
2020-02-03T19:49:19 DEBUG [P2P] Received message=merkleblock from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:49:19 DEBUG [P2P] Received merkle block, expecting 1 TX(s)
2020-02-03T19:49:19 DEBUG [P2P] Received message=tx from peer=Peer(localhost/127.0.0.1:18444) state=Normal 
2020-02-03T19:49:19 DEBUG [P2P] Transaction=DoubleSha256BDigestBE(50c635e2f4bb3ef5f5da21c9387583ce076d811c846aa766fe138766c8c9a6c6) matched merkleBlock=DoubleSha256BDigestBE(62e2f6cc19f171c781debd3f067b68dbdc435c30252fadeb964539d18857f22d)
2020-02-03T19:49:19 DEBUG [P2P] We've received all transactions (1) for merkleBlock=DoubleSha256BDigestBE(62e2f6cc19f171c781debd3f067b68dbdc435c30252fadeb964539d18857f22d)
```
In which it appears I am getting SPV instead of neutrino messages, and calling `getblockcount`, `getfilterheadercount`, `getfiltercount` yield old numbers.

If I then restart my node by killing the server and running it again, it then syncs the new blocks like normal and updates the numbers.

**Comment by nkohen** [comment]
I am running `binaries/bitcoind/bitcoin-0.18.99/bin/bitcoind` with a datadir that has a regtest `bitcoin.conf` and a port specified that I peer to in my `bitcoin-s.conf`

**Comment by benthecarman** [comment]
Closed by #1730
